### PR TITLE
Slowly migrate AMP to a state where we can obfuscate all properties.

### PR DESCRIPTION
**Pull request by cramforce**
Uses the type system to ensure that objects that must be treated as non-obfuscateable are using bracket notation for property access.

This is a first PR of many. Part of #9876

### CODE REVIEW

File: src/utils/object.js [hunk 1]
```diff
@@ -36,6 +36,17 @@ export function map(opt_initial) {
...
 
 /**
+ * @param {!Object=} opt_initial
+ * @return {!JsonObject}
+ */
+export function dict(opt_initial) {
+  if (opt_initial) {
+     return /** @type {!JsonObject} */ (opt_initial);
+  }
+  return /** @type {!JsonObject} */ ({});
```
Comment (erwinmombay) [inline src/utils/object.js]: should we do Object.create(null) here

Comment (cramforce) [inline src/utils/object.js]: Good call. Just made the function pass through to `map`

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -94,14 +94,14 @@ export class AmpAnalytics extends AMP.BaseElement {
     this.requests_ = {};
 
     /**
-     * @private {JSONType}
+     * @private {JsonObject}
      */
-    this.config_ = /** @type {JSONType} */ ({});
+    this.config_ = /** @type {JsonObject} */ ({});
```
Comment (erwinmombay) [inline extensions/amp-analytics/0.1/amp-analytics.js]: would it be better to call `dict()` here? (besides the implication of adding additional code to the binary)

Comment (cramforce) [inline extensions/amp-analytics/0.1/amp-analytics.js]: yes

**Comment by erwinmombay** [comment]
LGTM

**Pull request merged**